jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppClassLoader.java

```diff
@@ -290,14 +293,20 @@ public void addJars(Resource libs)
 
             try (Stream<Path> streamEntries = Files.list(dir))
             {
-                streamEntries
+                List<URI> entries = streamEntries
                     .filter(Files::isRegularFile)
                     .filter(this::isFileSupported)
                     .sorted(Comparator.naturalOrder())
                     .map(Path::toUri)
+                    .filter(new UniqueURIPredicate(getURLs())) // ignore URLs that are already present in classloader
```

Comment:
`UniqueURIPredicate` transforms the URIs into `jar:...`ones while this is only done at the following line `map(URIUtil::toJarFileUri)` here. Can you shed some light on this?

Reply:
Ah, good point, not worth doing that twice.

Reply:
I think if some tries to add the same jar twice, then so be it.

Reply:
If duplicate jars are properly mounted twice and unmounted twice, then I agree with @janbartel